abi/src/main/java/org/web3j/abi/TypeDecoder.java

```diff
@@ -278,4 +282,116 @@ static Utf8String decodeUtf8String(String input, int offset) {
                     e);
         }
     }
+
+    static BigInteger asBigInteger(Object arg) {
+        if (arg instanceof BigInteger) {
+            return (BigInteger) arg;
+        } else if (arg instanceof BigDecimal) {
+            return ((BigDecimal) arg).toBigInteger();
+        } else if (arg instanceof String) {
+            return Numeric.toBigInt((String) arg);
+        } else if (arg instanceof byte[]) {
+            return Numeric.toBigInt((byte[]) arg);
+        } else if (arg instanceof Number) {
+            return BigInteger.valueOf(((Number) arg).longValue());
+        }
+        return null;
+    }
+
+    static List arrayToList(Object array) {
+        int len = java.lang.reflect.Array.getLength(array);
+        ArrayList rslt = new ArrayList(len);
+        for (int i = 0; i < len; i++) {
+            rslt.add(java.lang.reflect.Array.get(array, i));
+        }
+        return rslt;
+    }
+
+    public static Type instantiateType(String solidityType, Object value) throws
+            InvocationTargetException, NoSuchMethodException, InstantiationException,
+            IllegalAccessException, ClassNotFoundException {
+        return instantiateType(makeTypeReference(solidityType), value);
+    }
+
+    public static Type instantiateType(TypeReference ref, Object value) throws
```

Comment:
I'd suggest to split this method into smaller ones to make it more readable.

Reply:
done. split into 2 methods: instantiateArrayType and instantiateAtomicType